### DISCUSSION TITLE
Unclear error message on registration and password reset for invalid password format

### DISCUSSION THREAD

**Discussion by Abdallah-Awwad** [directus/directus#24901] [Feature Requests]
### **Description:**

New invited users attempting to register for Data Studio encounter a generic "Unexpected Error" message when their password doesn't meet the "Auth Password Policy" requirements (e.g., minimum length of 8 characters). This is misleading as it suggests an internal server error rather than an issue with the password itself.

### Expected Behavior:
Upon entering an invalid formatted password, the user should receive a clear and specific error message detailing the password policy requirements. This could include:
- Minimum password length.
- Requirement for uppercase, lowercase characters, numbers and special characters.

### Impact:

The unclear error message creates a confusing experience for new users.
Users might mistakenly believe there's a server issue instead of an issue with their password format.
This can lead to frustration and difficulty completing the registration or password reset process.

### Additional Notes:

The same generic error message appears during password reset attempts when the new password doesn't meet the policy.
![image](https://github.com/directus/directus/assets/129249191/24b3b802-b072-475d-98a0-58b8b0793072)

### To Reproduce

1. Attempt to register for a new account or reset an existing password for Data Studio.
2. Access the registration or password reset link from your email.
3. During registration/reset, enter a password that violates at least one requirement of the "Auth Password Policy" (e.g., a password less than 8 characters long or lacking special characters).
4. Observe the error message displayed after attempting to submit the invalid password.

### Directus Version

10.10.4

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by br41nslug** [comment]
Leaving reference to this old discussion https://github.com/directus/directus/discussions/12788

**Comment by ghost** [comment]
I can't seem to find the upvote in this issue (and the old one is locked) but I consider this a basic _quality of life_ feature. These low code db manager systems are usually used by none technical people and this problem can be an adoption obstacle. I encounter this problem when adopting the tool at our organization. A basic "password policy error message" would be sufficient. It could have a stock message for each of the three template options and you could customize it if you create a custom one. I've read the discussion on showing checkboxes if the password complies with each rule, but this may only delay a basic still sufficient fix. I can help coding a solution.

**Comment by ioiste** [comment]
Still reproducible on Directus 11.17.4 as of August 2026 (also nothing in the 12.x changelog addresses it). Adding a concrete repro since this thread is mostly +1s.

**Setup:** self-hosted, `directus/directus:11`, Postgres 16. Project setting `auth_password_policy` = Strong. Users invited from the Studio UI (Users → Invite), so they land on `/admin/accept-invite`.

**Repro:**
1. Set Settings → Security → Password Policy to "Strong".
2. Invite a user from the Studio.
3. Open the invite link, enter a password that fails the policy (e.g. letters + digits, no special character).
4. Submit.

**Result:** the form shows exactly `Validation failed`. Nothing else — no mention of the password, no requirements, no hint about what to change.

**But the API already returns everything needed.** `POST /users/invite/accept` responds with:

```json
{
  "errors": [
    {
      "message": "Validation failed for field \"password\". Value doesn't have the correct format.",
      "extensions": {
        "field": "password",
        "path": [],
        "type": "regex",                                                                                                                                                       "invalid": "<the password the user typed>",
        "code": "FAILED_VALIDATION"
      }
    }
  ]
}                                                                                                                                                                      ```
```

The Studio discards it: errors are translated by `code`, so `FAILED_VALIDATION` → the generic `Validation failed` string, and the specific `message` never reaches the user. The information is one field away from being shown.

**Why this hurts more than a typical papercut:** this is the *first* screen a new user ever sees, before they have an account. They cannot get help from inside the product, they cannot see the policy, and they have no way to guess which of the four possible rules they violated. In practice they email an admin or give up. For
non-technical invitees this is a hard stop on onboarding.

**Requests, in increasing order of effort:**

1. Surface the API `message` for `FAILED_VALIDATION` when `extensions.field === "password"`, instead of the generic code translation. Smallest possible change, fixes the worst of it.
2. Better: translate the policy into a human sentence, e.g. "Password must be at least 8 characters and include an uppercase letter, a lowercase letter, a number, and a special character."
3. Best: show the requirements on the accept-invite and reset-password forms *before* submission, so the failure never happens.

**No self-hosted workaround exists.** `/admin/accept-invite` is a public app route and isn't overridable by an extension. The only options are intercepting the route at the reverse proxy to serve a custom page, or forking the app bundle — both disproportionate for a string.

**Separate, smaller issue in the same response:** `extensions.invalid` echoes the rejected password back in cleartext. It reaches the browser and lands in anything that logs API errors. Worth redacting for the password field regardless of how the message question is resolved.
